### ISSUE TITLE
[HOLD for 56812] [MEDIUM] iOS - Troubleshoot- The "Client side logging" toggle is automatically turned off

### ISSUE THREAD

**Issue by vincdargento** [Expensify/App#53713]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.72-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** N/A
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Launch the app
2. Navigate to the Troubleshoot in the Settings
3. Turn on the Client side logging toggle
4. Tap the View debug console button
5. Tap the Share log button
6. Select the chat and send a log
7. Return the Troubleshoot

## Expected Result:
The  "Client side logging" toggle is turned on.

## Actual Result:
After sharing a log, the "Client side logging" toggle is automatically turned off and can't be toggled on by the first try.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/65929546-fcaf-4e75-9c9e-465cf3d310b5

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866259215454556892</li>
        <li>Upwork Job ID: 1866259215454556892</li>
        <li>Last Price Increase: 2024-12-09</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @zfurtak</details>

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The client side logging toggle is switched off again after sharing the log.

### What is the root cause of that problem?
This is similar to https://github.com/Expensify/App/pull/50554. Based on the PR explanation, the style on the UI thread is lost. This happens when our FreezeWrapper freezes the screen. 
https://github.com/Expensify/App/blob/b595f5d24141a403fcc3f44c73d00cd6951dca7d/src/components/withPrepareCentralPaneScreen/index.native.tsx#L4-L10

In our case, the switch depends on the offsetX shared value to animate the style using `useAnimatedStyle`.
https://github.com/Expensify/App/blob/b595f5d24141a403fcc3f44c73d00cd6951dca7d/src/components/Switch.tsx#L38-L61

### What changes do you think we should make in order to solve the problem?

I think this is best to be fixed on the reanimated repo, but in case we want to fix it on our app, we can toggle the `offsetX` value on/off every time the screen is unfrozen. react-freeze uses Suspense which based on the [doc](https://react.dev/reference/react/Suspense#:~:text=If%20React%20needs,content%20is%20hidden.), it will trigger `useLayoutEffect` when unsuspended.
```
useLayoutEffect(() => {
    offsetX.set(!isOn ? OFFSET_X.ON : OFFSET_X.OFF);
    offsetX.set(isOn ? OFFSET_X.ON : OFFSET_X.OFF);
}, []);
```

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

N/A

**Comment by Christinadobrzyn** [comment]
Oh interesting! I think this should be fixed so adding an External label

**Comment by Christinadobrzyn** [comment]
waiting for proposals

**Comment by Christinadobrzyn** [comment]
Oh, we have a proposal! @getusha can you please [review](https://github.com/Expensify/App/issues/53713#issuecomment-2524900226)?

**Comment by getusha** [comment]
@Christinadobrzyn could we get someone from Software Mansion to have a look on this? seems like something we would want to fix in `react-native-reanimated` library

**Comment by Christinadobrzyn** [comment]
yep! asking - https://expensify.slack.com/archives/C04878MDF34/p1734105460130609

**Comment by zfurtak** [comment]
Hi I'm Zuza from Software Mansion, and I will take care of this issue 😊

**Comment by Christinadobrzyn** [comment]
Thanks for your help @zfurtak - I've added you to this issue.

**Comment by Christinadobrzyn** [comment]
Just a note for Melvin - We're working on a PR

**Comment by Christinadobrzyn** [comment]
Hi @zfurtak just checking to see if you have an update for us. Thank you!

**Comment by Christinadobrzyn** [comment]
Just a heads up that I'm going to be ooo until Dec 30th. I'm not going to assign this to a BZ teammate but if anything is urgent, please reach out to the team for a volunteer.

**Comment by zfurtak** [comment]
So far I didn't find the best solution and fyi I will be ooo until 10th January

**Comment by Christinadobrzyn** [comment]
Okay thanks for the update @zfurtak - I think we can keep this assigned for now since most of SWM is gone until that time.

**Comment by Christinadobrzyn** [comment]
I think we're good Melvin - @zfurtak is ooo until Jan 10th, I think we're going to stick with them as the contributor for this job so we'll continue when they are back.

**Comment by Christinadobrzyn** [comment]
This is on hold for now but will be worked on by @zfurtak soon!

**Comment by zfurtak** [comment]
I will continue working on it on Monday 😊

**Comment by zfurtak** [comment]
Hi, sorry but due to other issues I wasn't able to work on this one, I'm ooo until Wednesday but if it's any rush I can ask my colleagues from SWM to look at it.

**Comment by Christinadobrzyn** [comment]
I think if there is someone who can take this @zfurtak that might be good since it's been pending for over a month now. Let us know if you find someone!

**Comment by Christinadobrzyn** [comment]
Nudge @zfurtak are you able to grab this again?

**Comment by zfurtak** [comment]
Yes sure! I can continue, sorry for the delay...

**Comment by zfurtak** [comment]
hello, little update from me: I'm working on it with my colleagues who maintain reanimated, I prepared them a raw reproduction sample so then can look into the library's behaviour

**Comment by zfurtak** [comment]
I contacted my colleagues from reanimated and they're working on the fix, it's almost ready but to be sure they want to test it very well.

**Comment by Christinadobrzyn** [comment]
Thanks @zfurtak feel free to let us know if there's an update or if we should add anyone from your team to this issue for tracking!

**Comment by zfurtak** [comment]
Recently I tried to reproduce our issue, but due to big navigation changes I couldn't do it because of a navigation bug, which I reported to my colleagues and now [the PR ](https://github.com/Expensify/App/pull/57110)with fix is waiting for a review 😊
About the reanimated fix, as I heard the bug was quite complex and annoying, however they managed to raise [the PR ](https://github.com/software-mansion/react-native-reanimated/pull/7053)which is now also waiting for a review 👍

**Comment by mallenexpensify** [comment]
Thanks for being active on this issue @zfurtak .  I updated title to include HIGH since that's the status on #quality.  
Will keep eyes on this when doing weekly (or likely bi-weekly) reports for HIGH issues in #quality.

**Comment by zfurtak** [comment]
Hi 😄 The navigation fix I wrote about above was reverted, so for now it's not possible to reproduce this issue. However the reanimated fix was merged so I think that our problem will be resolved soon!! 💪

**Comment by zfurtak** [comment]
Hello :) from Monday I will be OOO until 20th of March, but I asked my colleagues to look after this issue, so I believe that someone will keep an eye on it 😊

**Comment by blazejkustra** [comment]
FYI, I'll keep an eye on this one!

**Comment by blazejkustra** [comment]
As I understand it, we are currently waiting on a fix for the navigation issue (which was previously reverted [here](https://github.com/Expensify/App/pull/57110)) before we can even reproduce **this** issue. 

Meanwhile a fix for **this** issue was already [merged](https://github.com/software-mansion/react-native-reanimated/pull/7053) upstream but not yet available in the newest version of Reanimated - so I'll observe all threads and if anything changes will let you know.

**Comment by blazejkustra** [comment]
I think [this](https://github.com/Expensify/App/issues/56812) is it 🙇

**Comment by WojtekBoman** [comment]
[A new PR](https://github.com/Expensify/App/pull/57753) fixing [this issue](https://github.com/Expensify/App/issues/56812) is ready

**Comment by Christinadobrzyn** [comment]
we're on hold for https://github.com/Expensify/App/issues/56812

**Comment by Christinadobrzyn** [comment]
It looks like https://github.com/Expensify/App/issues/56812 is in production. Can we retest or continue with this issue? @WojtekBoman @blazejkustra TY!

**Comment by Christinadobrzyn** [comment]
@zfurtak @getusha just checking on where we are with this?

**Comment by zfurtak** [comment]
Hello! I'm back 😃 I've just checked the app on ios and the bug is no longer there. So I assume that this issue is fixed! 💪
cc: @Christinadobrzyn

**Comment by mallenexpensify** [comment]
Was also unable to reproduce, let's close!
